src/test/rustdoc/intra-doc-link-generic-params.rs

```diff
@@ -0,0 +1,10 @@
+#![crate_name = "foo"]
+
+//! Here's a link to [`Vec<T>`] and one to [`Box<Vec<Option<T>>>`].
+//!
+//! And what about a link to just [`Option`] and, with the generic, [`Option<T>`]?
+
+// @has foo/index.html '//a/@href' 'https://doc.rust-lang.org/nightly/alloc/vec/struct.Vec.html'
+// @has foo/index.html '//a/@href' 'https://doc.rust-lang.org/nightly/alloc/boxed/struct.Box.html'
+// @has foo/index.html '//a/@href' 'https://doc.rust-lang.org/nightly/core/option/enum.Option.html'
+// @has foo/index.html '//a/@href' 'https://doc.rust-lang.org/nightly/core/option/enum.Option.html'
```

Comment:
This is not a sufficient test, it will pass even if `Option<T>` doesn't resolve. The simplest way is to just remove `[Option]` from line 5.
```suggestion
//! And what about a link to [`Option<T>`]?

// @has foo/index.html '//a/@href' 'https://doc.rust-lang.org/nightly/alloc/vec/struct.Vec.html'
// @has foo/index.html '//a/@href' 'https://doc.rust-lang.org/nightly/alloc/boxed/struct.Box.html'
// @has foo/index.html '//a/@href' 'https://doc.rust-lang.org/nightly/core/option/enum.Option.html'
// @has foo/index.html '//a/@href' 'https://doc.rust-lang.org/nightly/core/option/enum.Option.html'
```

Reply:
Can I leave the `Option` link if I add the link text to the `@has`?

Reply:
Sure, that would work.